packages/gatsby-link/src/index.js

```diff
@@ -109,11 +109,22 @@ class GatsbyLink extends React.Component {
       ...rest
     } = this.props
 
-    const LOCAL_URL = /^\/(?!\/)/
-    if (process.env.NODE_ENV !== `production` && !LOCAL_URL.test(to)) {
-      console.warn(
-        `External link ${to} was detected in a Link component. Use the Link component only for internal links. See: https://gatsby.app/internal-links`
-      )
+    if (process.env.NODE_ENV !== `production`) {
+      const LOCAL_URL = /^\/(?!\/)/
+      const message = `External link ${to} was detected in a Link component. Use the Link component only for internal links. See: https://gatsby.app/internal-links`
+
+      try {
+        const link = new URL(to, window.location)
```

Comment:
This is problematic because it'll always revert to the old behaviour (inside the `catch`) during server side rendering since `window` will not be accessible